sdks/python/apache_beam/transforms/window.py

```diff
@@ -220,7 +227,9 @@ def __hash__(self):
     return hash((self.start, self.end))
 
   def __eq__(self, other):
-    return self.start == other.start and self.end == other.end
+    return type(self) == type(other) and \
```

Comment:
Same comment about type mismatch.

Reply:
Can we put the whole test in parens and avoid backslash line termination as recommended in PEP8.  The problem with backslash line termination is that s whitespace character to the right of the backslash break the script but is invisible to the reader.

Reply:
+1